### PR TITLE
2015-day-20-part-1

### PR DESCRIPTION
**Pull request by Herrera93**
This library counts the presents delivered to a house by elves.
At first I wanted to get all the factors of a house number, multiply them by 10 and add them up. This method is still present in the library function `presents_in_house` but this approach was very inefficient when trying to find the house number with `<= number of presents` so I did instead is simulate the houses in a `Vec` and had the elves add the presents in the `Vec` once that was done I just had to find the lowest index with less than or equal to the number of presents.

### CODE REVIEW

**Comment by Herrera93** [comment]
[2015-day-20-part-1](https://trello.com/c/Kmup3ycL/23-2015-day-20-part-1)

File: src/year2015/day_20.rs [hunk 1]
```diff
@@ -0,0 +1,68 @@
+use std::collections::HashMap;
+
+pub fn prime_sieve(n: i32) -> Vec<i32> {
```
Comment (kanekv) [inline src/year2015/day_20.rs]: does it have to be public?

File: src/year2015/day_20.rs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            for i in (p * 2..=n).step_by(p as usize) {
+                prime[i as usize] = false;
+            }
+        }
+        p += 1;
+    }
+    (2..=n).filter(|p| prime[*p as usize]).collect()
+}
+
+pub fn get_factors(n: i32) -> Vec<i32> {
```
Comment (kanekv) [inline src/year2015/day_20.rs]: does this have to be public?

File: src/year2015/day_20.rs (hunk 1 above)
Comment (Herrera93) [reply src/year2015/day_20.rs]: Oh you're right, I made it public to test it. Let me change both of them back to private.

Comment (Herrera93) [reply src/year2015/day_20.rs]: I'm also going to add `#[ignored]` to `test_day_20_part_1` since I got 2.293927535 secs on it.

Comment (kanekv) [reply src/year2015/day_20.rs]: @Herrera93 let's not have tests that run more than a couple of hundreds ms.

**Pull request merged**